packages/docregistry/src/context.ts

```diff
@@ -67,7 +66,7 @@ export class Context<
     this._model = this._factory.createNew({
       languagePreference: lang,
       sharedModel,
-      collaborationEnabled: PageConfig.getOption('collaborative') === 'true'
+      collaborationEnabled: sharedFactory === null ? false : true
```

Comment:
I would keep the usage of `PageConfig.getOption('collaborative') === 'true'`. Mainly because providing a shared model factory does not ensure live collaboration is enabled.

Reply:
I am not sure it is a good idea to keep the `PageConfig.getOption('collaborative') === 'true'`. That means we will have to set that attribute from the extension, which will be global. We could have two drives installed, one collaborative and the other not. Two components can influence whether a document is collaborative:

1- A provider that supports RTC (the YDrive).
2- And the DocumentModel is designed to support RTC.

We know the drive supports collaboration when it returns a shared model for a given path:
```
const sharedFactory = this._manager.contents.getSharedModelFactory(
      this._path
);
```

And we know the DocumentModel supports collaboration by checking the `collaborative` property in its factory, `DocumentModelFactory.collaborative === true`.

To activate RTC for a given path, we check whether both components support RTC and set the property `DocumentModel.collaborative` for each document.

Maybe we should make the IDrive more explicit by adding a `collaborative` property instead of assuming the model is collaborative when it returns a shared model.

Reply:
The multiple drive cases is a good argument to drop `PageConfig`. But still having a `sharedFactory` is not, we know it is what we will use for `jupyterlab_rtc`. But we should not make assumption on what other may do with it. So I like to go in the direction of your proposal. But instead of setting it at the drive level that is deeply hidden, I would either 
- extend `ISharedFactory` to provide a `collaborative` flag
- move the `collaborative` attribute from `DocumentRegistry.IModel` to `ISharedDocument` (this would be valid as `ISharedFactory` receive all information to create the `ISharedDocument`).

I have a small preference for the second approach. What do you think?

Reply:
I like it!

Reply:
I changed my mind, lets's go with the first option. It will be easier to maintain.

Reply:
We went for the first option because in the case of the second, we were not happy about adding a flag in `@jupyter/ydoc` that would have been specific to the JupyterLab infrastructure.